src/backend/functions/projects-edit.ts

```diff
@@ -0,0 +1,528 @@
+/*
+ * This file is part of NER's PM Dashboard and licensed under GNU AGPLv3.
+ * See the LICENSE file in the repository root folder for details.
+ */
+
+import middy from '@middy/core';
+import jsonBodyParser from '@middy/http-json-body-parser';
+import httpErrorHandler from '@middy/http-error-handler';
+import validator from '@middy/validator';
+import { Handler } from 'aws-lambda';
+import { Description_Bullet, PrismaClient } from '@prisma/client';
+import { buildSuccessResponse, DescriptionBullet, projectEditInputSchemaBody } from 'utils';
+
+const prisma = new PrismaClient();
+
+export const descBulletConverter = (descBullet: Description_Bullet): DescriptionBullet => ({
+  ...descBullet,
+  id: descBullet.descriptionId ?? undefined,
+  dateDeleted: descBullet.dateDeleted ?? undefined
+});
+
+export const editProject: Handler = async ({ body }, _context) => {
+  const {
+    wbsElementId,
+    crId,
+    userId,
+    budget,
+    summary,
+    rules,
+    goals,
+    features,
+    otherConstraints,
+    name,
+    wbsElementStatus
+  } = body;
+
+  // get the original project so we can compare things
+  const originalProject = await prisma.project.findUnique({
+    where: {
+      wbsElementId
+    },
+    include: {
+      wbsElement: true,
+      goals: true,
+      features: true,
+      otherConstraints: true
+    }
+  });
+
+  // if it doesn't exist we error
+  if (originalProject === null) {
+    throw new TypeError('Project with given wbsElementId does not exist!');
```

Comment:
This should be using `return buildClientFailureResponse('msg');` instead of throwing a TypeError. I know it's just copy-pasting from existing code, but basically a) this isn't the correct scenario for a type error and b) throwing an error in the back-end makes for terrible and difficult to decipher error messages.